gc/aesndlib.h

```diff
@@ -35,17 +35,18 @@
 
 typedef struct aesndpb_t AESNDPB;
 
-typedef void (*AESNDVoiceCallback)(AESNDPB *pb,u32 state);
-typedef void (*AESNDAudioCallback)(void *audio_buffer,u32 len);
+typedef void (*AESNDVoiceCallback)(AESNDPB *pb,u32 state,void *cbArg);
+typedef void (*AESNDAudioCallback)(void *audio_buffer,u32 len,void *cbArg);
 
 void AESND_Init(void);
 void AESND_Reset(void);
 void AESND_Pause(bool pause);
 u32 AESND_GetDSPProcessTime(void);
 f32 AESND_GetDSPProcessUsage(void);
+AESNDAudioCallback AESND_RegisterAudioCallbackWithArg(AESNDAudioCallback cb, void *cbArg);
 AESNDAudioCallback AESND_RegisterAudioCallback(AESNDAudioCallback cb);
 
-AESNDPB* AESND_AllocateVoice(AESNDVoiceCallback cb);
+AESNDPB* AESND_AllocateVoice(AESNDVoiceCallback cb, void *cbArg);
```

Comment:
doesn't this function also need an overload?

Reply:
Fixed. By the way, would AESNDVoice/AudioCallback need an alternate Arg form? Old callbacks that lack a void* parameter might break.

Reply:
i was also thinking about it just now. i *think* it should be fine. the code will have the registers set up for the function and have their values on the stack before the callback is called. the callback can do whatever the fuck it want, even overwrite the register. once its done the values are popped back off the stack, and its like nothing happened.
the whole of IOS/Syscalls work on this principal.
it would be an issue if the callback had more arguments than the caller intended, but this is not the case here.
this would also be a problem ,afaik, if the function had more parameters than registers so parameters would need to be on the stack, but PPC has like... 31 registers which 8 are for function parameters so not gonna happen here i think hehe

verify it with a dol/elf and throwing it against ghidra to see if im right or not.

also, its the breaking changes and issues that we are trying to resolve here :)

Reply:
Ok. So should I make alternate Arg variants for the two types?

Reply:
Are there any other changes I should make?